### DISCUSSION TITLE
Linux 编译版本，文本渲染是否支持 emoji

### DISCUSSION THREAD

**Discussion by changaolee** [Tencent/libpag#2554] [Q&A] [answered]
文本中 emoji 会变成问号，使用 [NotoColorEmoji.ttf](https://github.com/googlefonts/noto-emoji/raw/main/fonts/NotoColorEmoji.ttf) 字体也无法渲染。

**Comment by changaolee** [comment]
补充下：[NotoColorEmoji.ttf](https://github.com/googlefonts/noto-emoji/raw/main/fonts/NotoColorEmoji.ttf) 字体是配置在 fallback 里的

**Comment by 0x1306a94** [comment]
@domchen 这个有解决方案了么？
* 原来用的4.3.56 版本在Linux上渲染出来如下

<img width="588" alt="iShot_2025-05-06_19 25 25" src="https://github.com/user-attachments/assets/73585207-8dae-445e-aecb-af6927cec89d" />

* 刚刚更新到 4.4.29 后
* 在开发的Ubuntu 22.04 上渲染结果如下

<img width="1164" alt="iShot_2025-05-06_19 25 00" src="https://github.com/user-attachments/assets/cb21935d-e4fc-47b8-ad14-55800fd431b0" />

* 而在实际的k8s 部署后渲染emoji 直接被忽略了。同样也是Ubuntu 22.04

<img width="1211" alt="iShot_2025-05-06_19 28 22" src="https://github.com/user-attachments/assets/5d9f6298-072f-45a2-802d-3f54a7aa42b0" />

**Comment by kevingpqi123** [comment]
关于文本渲染，着重看下这篇文档： 
https://pag.io/docs/api-instructions.html#pag-%E5%AD%97%E4%BD%93%E6%B3%A8%E5%86%8C

pag 在 linux 平台提供了 demo， 可以带入验证
https://github.com/Tencent/libpag/blob/main/linux/src/main.cpp

如果有问题，将复现 demo 提供出来

关于提问题的建议： 不要仅仅描述一个现象，将你们要实现的功能、怎么实现的、哪里不符合预期都提供出来，一次性就可以获取到答案，为了提高对接问题的效率，我们会优先回复提供信息全面的问题

**Comment by 0x1306a94** [comment]
@kevingpqi123 我将 https://github.com/Tencent/libpag/blob/main/resources/font/NotoColorEmoji.ttf 这个字体文件通 SetFallbackFontPaths 注册后。可以渲染出emoji了。我将Apple平台`/System/Library/Fonts/Apple Color Emoji.ttc`这个字体文件放到Linux上然后用SetFallbackFontPaths注册也可以了。

* `NotoColorEmoji.ttf`效果

<img width="537" alt="截屏2025-05-06 22 16 55" src="https://github.com/user-attachments/assets/1caf77c4-5d7a-4c90-82b8-e6e0402183d5" />

* `Apple Color Emoji.ttc` 在Linux上的效果对齐有点问题

<img width="536" alt="截屏2025-05-06 22 17 11" src="https://github.com/user-attachments/assets/eb11e0de-0952-40d1-8dd4-1f275c7e03fa" />

* 这是在macOS上运行渲染的效果
<img width="527" alt="截屏2025-05-06 22 20 23" src="https://github.com/user-attachments/assets/e2d93ba7-89e2-47d2-88a0-8dcb6cc82cae" />

**Reply by domchen** [reply, marked as answer]
应该是因为你用了谷歌最新的矢量emoji字体，换成安卓之前版本的位图emoji字体就行，也就是我们仓库里带的那个。苹果的emoji也是位图的。目前还没有去适配最新的矢量彩色字体格式，这个会导致包体增加会大非常多。
